api/src/clojure/org/akvo/flow_api/endpoint/sync.clj

```diff
@@ -41,20 +41,24 @@
                 db-spec (-> deps :unilog-db :spec (assoc :db-name db-name))
                 offset (Long/parseLong cursor)]
             (if (unilog/valid-offset? offset db-spec)
-              (let [changes (->
-                             (unilog/process-unilog-events offset db-spec instance-id (:remote-api deps))
-                             (select-keys [:form-instance-changed
-                                           :form-instance-deleted
-                                           :form-changed
-                                           :form-deleted
-                                           :data-point-changed
-                                           :data-point-deleted
-                                           :unilog-id])
-                             (update :form-deleted #(map str %))
-                             (update :form-instance-deleted #(map str %))
-                             (update :data-point-deleted #(map str %)))]
-                (response {:changes (dissoc changes :unilog-id)
-                           :next-sync-url (next-sync-url (utils/get-api-root req) alias (:unilog-id changes))}))
+              (if (= offset (unilog/get-cursor db-spec)) ;; end of the log
```

Comment:
This is the third JDBC query: 1 for valid, 1 for last cursor and 1 for reading the data. 

Not an issue if we open the DB connection once, but if we open/close for each, maybe we should look into avoiding 2 of those 3 calls.

Being very picky here, so feel free to ignore.

Reply:
* the last cursor is one code path (`initial=true`)
* the valid and reading (`next=true`), for checking valid and reading data can be transformed into one